scripts/migration/migrate-istio

```diff
@@ -0,0 +1,1360 @@
+    fatal "Configuration backup not found, please run pre-migration step and retry installation."
+  fi
+}
+
+configure_package() {
+  info "Configuring kpt package..."
+
+  run kpt cfg set asm gcloud.container.cluster "${CLUSTER_NAME}"
+  run kpt cfg set asm gcloud.core.project "${PROJECT_ID}"
+  run kpt cfg set asm gcloud.project.environProjectNumber "${FLEET_PROJECT_NUMBER}"
+  run kpt cfg set asm gcloud.compute.location "${CLUSTER_LOCATION}"
+  run kpt cfg set asm gcloud.compute.network "${GCE_NETWORK_NAME}"
+  run kpt cfg set asm anthos.servicemesh.rev "${REVISION_LABEL}"
+  run kpt cfg set asm anthos.servicemesh.tag "${ASM_VERSION}"
+  run kpt cfg set asm anthos.servicemesh.hubTrustDomain "${FLEET_PROJECT_ID}.svc.id.goog"
+  HUB_IDP_URL="$(kubectl get memberships.hub.gke.io membership -o=jsonpath='{.spec.identity_provider}' || true)"
+  run kpt cfg set asm anthos.servicemesh.hub-idp-url "${HUB_IDP_URL}"
+  run kpt cfg set asm anthos.servicemesh.trustDomainAliases \
+    "${FLEET_PROJECT_ID}.svc.id.goog" "${FLEET_PROJECT_ID}.hub.id.goog"
+  info "Configured."
+}
+
+install_asm() {
+  local PARAMS
+  PARAMS="-f ${OPERATOR_MANIFEST}"
+  PARAMS="${PARAMS} -f ${CNI_MANIFEST}"
+  if [[ "${UPGRADE_GATEWAYS}" -eq 0 ]]; then
+    PARAMS="${PARAMS} -f ${NO_DEFAULT_INGRESS}"
+  fi
+  PARAMS="${PARAMS} -c ${KUBECONFIG}"
+  PARAMS="${PARAMS} --set revision=${REVISION_LABEL}"
+  PARAMS="${PARAMS} --skip-confirmation"
+
+  info "Installing ASM control plane..."
+  # shellcheck disable=SC2086
+  retry 2 run istioctl install $PARAMS
+
+  # Prevent the stderr buffer from ^ messing up the terminal output below
+  sleep 1
+
+  local RAW_YAML; RAW_YAML="${REVISION_LABEL}-manifest-raw.yaml"
+  local EXPANDED_YAML; EXPANDED_YAML="${REVISION_LABEL}-manifest-expanded.yaml"
+
+  PARAMS="-f ${OPERATOR_MANIFEST}"
+  if [[ "${UPGRADE_GATEWAYS}" -eq 0 ]]; then
+    PARAMS="${PARAMS} -f ${NO_DEFAULT_INGRESS}"
+  fi
+
+  # shellcheck disable=SC2086
+  run istioctl profile dump ${PARAMS} >| "${RAW_YAML}"
+  run istioctl manifest generate \
+    <"${RAW_YAML}" \
+    >|"${EXPANDED_YAML}"
+
+  run kubectl apply -f "${CANONICAL_CONTROLLER_MANIFEST}"
+  retry 2 run kubectl wait --for=condition=available --timeout=600s \
+      deployment/canonical-service-controller-manager -n asm-system
+
+  info ""
+  info "*******"
+  info "Control plane installation complete!"
+}
+
+init_meshconfig() {
+  if [[ "${FLEET_PROJECT_ID}" != "${PROJECT_ID}" ]]; then return; fi
+
+  info "Initializing meshconfig API..."
+  run curl --request POST --fail \
+  --data '' -o /dev/null \
+  "https://meshconfig.googleapis.com/v1alpha1/projects/${PROJECT_ID}:initialize" \
+  -K <(auth_header "$(get_auth_token)")
+}
+
+get_auth_token() {
+  local TOKEN; TOKEN="$(retry 2 gcloud --project="${PROJECT_ID}" auth print-access-token)"
+  echo "${TOKEN}"
+}
+
+auth_header() {
+  local TOKEN; TOKEN="${1}"
+  echo "--header \"Authorization: Bearer ${TOKEN}\""
+}
+
+relabel_for_asm() {
```

Comment:
nit: Could this accept an argument to relabel a particular namespace for canary based migration? 
Also, migration of namespaces could also be done using revision tags, refer https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages/blob/master/scripts/migration/migrate-addon#L355

Reply:
Updated to do this namespace by namespace